python/amici/ode_export.py

```diff
@@ -833,17 +834,24 @@ def import_from_sbml_importer(self,
             imported SBML model
         """
 
+        # get symbolic expression from SBML importers
         symbols = copy.copy(si.symbols)
 
-        # setting these equations prevents native equation generation
-        self._eqs['dxdotdw'] = si.stoichiometric_matrix
-        self._eqs['w'] = si.flux_vector
-        self._syms['w'] = sp.Matrix(
-            [sp.Symbol(f'flux_r{idx}', real=True)
-             for idx in range(len(si.flux_vector))]
-        )
-        self._eqs['dxdotdx'] = sp.zeros(si.stoichiometric_matrix.shape[0])
-
+        # assemble fluxes and add them as expressions to the model
+        fluxes = []
+        for ir, flux in enumerate(si.flux_vector):
```

Comment:
we don't necessarily need that, but could use the code lines from line 916 (`for symbol in [s for s in symbols if s != 'my']:`) and following (as I suggested). What's the purpose of changing that to this version?

Reply:
The idea here is to directly loop over `si.flux_vector` and only do that once. Directly looping over the vector entries is generally more pythonic, easier to mantain and less error-prone. Keeping everything in the same place is in the same spirit. Making changes is way easier as, you don't have to find all the different loops that may have drifted apart due to additional code changes (also less likely to miss one of them).

Initialization of expressions could indeed be done via the symbols variable, no strong preference on that.